### PR TITLE
Repeat check for frr config file

### PR DESCRIPTION
**Pull request by zeeke**
Hi,

I discovered a flaky unit test caused by the [asynchronous nature of the debouncer](https://github.com/metallb/metallb/blob/46675f03d788549897ff59c6b3b268c162a394d0/internal/bgp/frr/config.go#L328).
Here the symptom:
https://github.com/metallb/metallb/runs/4920413290?check_suite_focus=true

The check is done against an intermediate configuration file generated during the test, as `TestTwoAdvertisementsTwoSessions` triggers the configuration render twice.

This PR make the assertion to be repeated using a poll interval, so if spurious config file should be discarder.

Please, let me know if there is a better way to work around this edge case.

### CODE REVIEW

**Comment by liornoy** [comment]
I had this unit test flake on my PR also, so I'm glad you filed this fix and it's looking good IMO

File: internal/bgp/frr/frr_test.go [hunk 1]
```diff
@@ -31,9 +32,27 @@ func testOsHostname() (string, error) {
 
 func testCompareFiles(t *testing.T, configFile, goldenFile string) {
 	cmd := exec.Command("diff", configFile, goldenFile)
-	output, err := cmd.Output()
+	var lastError error
+
+	// Try comparing files multiple times because tests can generate more than one configuration
+	err := wait.PollImmediate(50*time.Millisecond, 2*time.Second, func() (bool, error) {
```
Comment (fedepaol) [inline internal/bgp/frr/frr_test.go:38]: Can we embed the check for the file presence here too?
Or, let's put the other way: with this, do we still need to wait for the file to be present?

File: internal/bgp/frr/frr_test.go [hunk 2]
```diff
@@ -31,9 +32,27 @@ func testOsHostname() (string, error) {
...
+			lastError = fmt.Errorf("command %s returned error: %s\n%s", cmd.String(), err, output)
+			return false, nil
+		}
+
+		return true, nil
+	})
+
 	if err != nil {
-		t.Fatalf("command %s returned error: %s\n%s", cmd.String(), err, output)
+		t.Fatalf("Failed to compare configfiles %s, %s, err %v", configFile, goldenFile, err)
```
Comment (fedepaol) [inline internal/bgp/frr/frr_test.go:51]: this won't dump lastError because fatal will halt the execution I believe?

**Comment by fedepaol** [comment]
Another note: I know that sleeping is not elegant, but I think we'd also risk to produce incomplete golden files for the same reason (and in that case, we won't have anything to compare against).
Would it make sense to put a sleep there, with a comment explaining why?

File: internal/bgp/frr/frr_test.go (hunk 1 above)
Comment (zeeke) [reply internal/bgp/frr/frr_test.go:38]: Good point! going to remove the file presence check, as it is embedded in the diff command

**Comment by fedepaol** [comment]
typo in the commit message: "asynchronousl"

File: internal/bgp/frr/frr_test.go [hunk 3]
```diff
@@ -30,15 +30,38 @@ func testOsHostname() (string, error) {
...
+		}
+
+		return true, nil
+	})
+
+	// err can only be a ErrWaitTimeout, as the check function always return nil errors.
+	// So lastError is always set
 	if err != nil {
-		t.Fatalf("command %s returned error: %s\n%s", cmd.String(), err, output)
+		t.Fatalf("failed to compare configfiles %s, %s using poll interval\nlast error: %v", configFile, goldenFile, lastError)
```
Comment (fedepaol) [inline internal/bgp/frr/frr_test.go:53]: I'd remove "using poll interval" from the log

**Comment by fedepaol** [comment]
LGTM, just the commit message needs to be fixed
